src/System.Device.Gpio/System/Device/I2c/I2cConnectionSettings.cs

```diff
@@ -39,5 +62,15 @@ internal I2cConnectionSettings(I2cConnectionSettings other)
         /// The bus address of the I2C device.
         /// </summary>
         public int DeviceAddress { get; }
+
+        public int SdaPin
```

Comment:
I also don't agree too much with this. One problem with this type that we learned in hindsight was that it was leaking implementation details for a very specific implementation (hardware I2c). Adding these would now also leak software implementation. We should talk about  this more on how to better approach the connection settings.